tests/test_core/test_alignment.py

```diff
@@ -2571,6 +2571,19 @@ def test_add_from_ref_aln(self):
             ValueError, aln1.add_from_ref_aln, aln2_wrong_refseq
         )  # test wrong_refseq
 
+    def test_annotate_matches_to(self):
```

Comment:
construct sample sequence to have multiple matches, e.g. pattern of "CCRC", instances of "CCAC" and "CCGC". Evaluate `allow_multiple` to be either True or False and the returned instances correspond to the correct "slices".